File: examples/swig-samples/python/test_nb_pub.py
#!/usr/bin/env python

#-
# Copyright (C) 2012  Oy L M Ericsson Ab, NomadicLab
#
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License version 2 as
# published by the Free Software Foundation.
#
# Alternatively, this software may be distributed under the terms of the
# BSD license.
#
# See LICENSE and COPYING for more details.
#

"""Very simple non-blocking text publisher example."""

from blackadder.blackadder import *

def _main(argv=[]):
    strategy = NODE_LOCAL
    if len(argv) >= 2:
        strategy = int(argv[1])
    
    ba = NB_Blackadder.Instance(True)

    @blackadder_event_handler
    def event_handler(ev):
        print ev
        if not ev or ev.type != START_PUBLISH:
            return
        data = "Hello"
        ba.publish_data(sid+rid, strategy, None, to_malloc_buffer(data))
    
    ba.setPyCallback(event_handler)
    
    sid0 = ""
    sid  = '\x0a'+6*'\x00'+'\x0b'
    rid  = '\x0c'+6*'\x00'+'\x0d'
    
    ba.publish_scope(sid, sid0, strategy, None)
    ba.publish_info(rid, sid, strategy, None)
    
    try:
        ba.join()
    finally:
        ba.disconnect()

if __name__ == "__main__":
    import sys
    _main(sys.argv)

File: examples/swig-samples/ruby/test_nb_pub.rb
#!/usr/bin/env ruby

#-
# Copyright (C) 2011-2012  Oy L M Ericsson Ab, NomadicLab
#
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License version 2 as
# published by the Free Software Foundation.
#
# Alternatively, this software may be distributed under the terms of the
# BSD license.
#
# See LICENSE and COPYING for more details.
#

# Simple text publisher example.

require 'blackadder'

ba = Blackadder::NB_Blackadder::Instance(true)

sid = 0xa.chr + 0x0.chr*6 + 0xb.chr
rid = 0xc.chr + 0x0.chr*6 + 0xd.chr

class EventHandler
    def initialize(ba, sid, rid)
        @ba = ba
        @sid = sid
        @rid = rid
    end
    def event_handler(ev)
        p ev
        p ev.type
        if ev != nil && ev.type == Blackadder::START_PUBLISH
            data = Blackadder::to_malloc_buffer("test")
            @ba.publish_data(@sid+@rid, Blackadder::NODE_LOCAL, nil, data)
        end
    end
end
evh = EventHandler.new(ba, sid, rid)

ba.setRubyCallback(evh, 'event_handler')

ba.publish_scope(sid, "", Blackadder::NODE_LOCAL, nil)
ba.publish_info(rid, sid, Blackadder::NODE_LOCAL, nil)

ba.join()

ba.disconnect()
